packages/rpc-types/src/__typetests__/lamports-typetests.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    FixedSizeCodec,
+    FixedSizeDecoder,
+    FixedSizeEncoder,
+    ReadonlyUint8Array,
+    VariableSizeCodec,
+    VariableSizeDecoder,
+    VariableSizeEncoder,
+} from '@solana/codecs-core';
+
+import {
+    getLamportsCodec,
+    getLamportsCodecFromCodec,
+    getLamportsDecoder,
+    getLamportsDecoderFromDecoder,
+    getLamportsEncoder,
+    getLamportsEncoderFromEncoder,
+    LamportsUnsafeBeyond2Pow53Minus1,
+} from '../lamports';
+
+// Default encoder
+{
+    const encoder = getLamportsEncoder();
+    encoder satisfies FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1>;
+    encoder.fixedSize satisfies number;
+    encoder.encode(1n as LamportsUnsafeBeyond2Pow53Minus1) satisfies ReadonlyUint8Array;
+}
+
+// Fixed size inner encoder
+{
+    const innerEncoder = {} as FixedSizeEncoder<bigint | number>;
```

Comment:
Could we just add an explicit fixed size here to make sure it's also not lost?

```suggestion
    const innerEncoder = {} as FixedSizeEncoder<bigint | number, 42>;
```